### PR TITLE
Feature: auto trigger snippets

### PR DESCRIPTION
**Pull request by yonip23**

### CODE REVIEW

File: src/inlineSuggestions/snippets/autoTriggerHandler.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+export default async function snippetAutoTriggerHandler({
+  document,
+  contentChanges,
+}: TextDocumentChangeEvent): Promise<void> {
+  const [change] = contentChanges;
+  const position = getCurrentPosition(change);
+  if (
+    !isInSnippetInsertion() &&
+    change.text.includes(EOL) &&
```
Comment (dimacodota) [inline src/inlineSuggestions/snippets/autoTriggerHandler.ts:19]: to readable  variable

File: src/inlineSuggestions/snippets/autoTriggerHandler.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    document.lineAt(position.line).text.trim() === ""
```
Comment (dimacodota) [inline src/inlineSuggestions/snippets/autoTriggerHandler.ts:20]: to readable  variable

File: src/inlineSuggestions/snippets/autoTriggerHandler.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  ) {
+    await requestSnippet(document, position, SnippetRequestTrigger.Auto);
+  }
+}
+
+function getCurrentPosition(change: TextDocumentContentChangeEvent): Position {
```
Comment (dimacodota) [inline src/inlineSuggestions/snippets/autoTriggerHandler.ts:26]: extract it and use it in `textListener`

Comment (yonip23) [reply src/inlineSuggestions/snippets/autoTriggerHandler.ts:26]: its not the same calculation... for snippets the current position is calculated differently

**Pull request merged**